app/routes/register.tsx

```diff
@@ -61,13 +65,24 @@ export const action: ActionFunction = async ({ request }) => {
       try {
         logger.info(`calling oneClick with ${apiKey}`);
 
-        const result = await oneClick(
-          apiKey as string,
+        const options: Partial<OneClickOptions> = {
           phone,
-          isRedirect
-            ? { redirectUrl, verificationOptions: 'only_code' }
-            : { redirectUrl }
-        );
+          birthDate,
+          redirectUrl,
+        };
+
+        // If the one-click non-hosted feature is enabled, set the isHosted option to false.
+        if (config.oneClickNonHostedEnabled) {
+          options.isHosted = false;
+        }
+
+        // If the one-click non-hosted feature is enabled or redirect query param is set,
+        // then set the verificationOptions to only_code.
+        if (isRedirect || config.oneClickNonHostedEnabled) {
```

Comment:
Unsure if this conditional is what we want. If `oneClickNonHostedEnabled` is true then there ought to be no redirect in the same sense as the other demo flavors.

Reply:
Per what was discussed in Slack, there are no differences for the non-hosted in that case. It should always request user OTP.